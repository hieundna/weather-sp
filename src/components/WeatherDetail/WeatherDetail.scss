.weather-detail {
    display: flex;

    .col-detail {
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: flex-start;
        width: 25%;
        color: #fff;
        border-width: 1px;
        border-style: solid;
        border-image:
            linear-gradient(to top,
                white,
                rgba(0, 0, 0, 0)) 1 100%;

        div {
            margin: 5px 0;
        }

        .title-detail {
            font-weight: 600;
        }

        .detail-value {
            padding: 8px;
            font-size: 24px;
            font-weight: 500;
            width: 30px;
            height: 30px;
            text-align: center;
        }

        .psi {
            background-color: #32a848;
            border-radius: 4px;
        }

        .circle-icon {
            width: 40px;
            height: 40px;
            margin-top: 15px;
            border: 2px solid #fff;
            border-radius: 100%;
            background-color: #94d6f7;
        }

        .add-text {
            font-size: 20px;
            font-weight: 500;
        }
    }

    .col-detail:nth-last-child(1) {
        justify-content: center;
        color: black;
    }
}

@media screen and (max-height: 694px) {
    .weather-detail {
        .col-detail {
            div {
                margin: 2px 0;
            }

            .detail-value {
                padding: 5px;
                font-size: 17px;
                width: 22px;
                height: 22px;
            }

            .circle-icon {
                width: 33px;
                height: 33px;
                margin-top: 8px;
            }

            .add-text {
                font-size: 17px;
            }
        }
    }
}