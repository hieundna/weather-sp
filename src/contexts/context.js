import { createContext, useCallback, useContext, useReducer } from 'react';
import { rootReducer, initialState } from './reducers/rootReducer'

const Context = createContext({});

export const AppProvider = props => {
    const { children } = props;
    const [data, dispatch] = useReducer(rootReducer, initialState);
    return (
        <Context.Provider value={{ data, setData: dispatch }}>
            {children}
        </Context.Provider>
    )
}

export const useAppContext = () => {
    const { data, setData } = useContext(Context)

    const dispatch = useCallback(
        ({ type, payload }) => {
            setData({ payload, type })
        },
        [setData]
    )
    return [data, dispatch]
}

export default Context;