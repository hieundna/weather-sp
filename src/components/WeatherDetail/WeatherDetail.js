import '../../styles/WeatherDetail.scss';
import { useAppContext } from '../../contexts/context'
// import { useContext } from 'react';

function WeatherDetail() {
    const [{weather}] = useAppContext();
    
    return (
        <div className='weather-detail'>
            <div className='col-detail'>
                <div className='title-detail'> PSI</div>
                <div className='detail-value psi'>{parseInt(weather?.main?.feels_like || 23)}</div>
                <div className='detail-state'>Good</div>
            </div>
            <div className='col-detail'>
                <div className='title-detail'> RAIN</div>
                <div className='detail-value'>{weather && weather.rain ? weather['rain']['1h'] : 0}</div>
                <div className='detail-state'>mm</div>
            </div>
            <div className='col-detail'>
                <div className='title-detail'> DENGUE</div>
                <div className='circle-icon'></div>
            </div>
            <div className='col-detail'>
                <div className='add-icon'><i className="far fa-plus-circle fa-2x"></i></div>
                <div className='add-text'>Add</div>
            </div>
        </div>
    );
}

export default WeatherDetail;
