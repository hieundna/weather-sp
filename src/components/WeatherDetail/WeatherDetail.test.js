import React from 'react';
import WeatherDetail from './WeatherDetail.js';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { AppProvider } from '../../contexts/context'

const wrapper = mount(<AppProvider> <WeatherDetail /> </AppProvider>);
describe('WeatherDetail Component', () => {
    test("expect to render WeatherDetail component", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    })
    test('renders without error', () => {
        const weatherDetailComponent = wrapper.find(".col-detail");
        expect(weatherDetailComponent.length).toBe(4);
    });

})