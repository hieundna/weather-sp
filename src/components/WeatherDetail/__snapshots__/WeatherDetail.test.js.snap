// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherDetail Component expect to render WeatherDetail component 1`] = `
<AppProvider>
   
  <WeatherDetail>
    <div
      className="weather-detail"
    >
      <div
        className="col-detail"
      >
        <div
          className="title-detail"
        >
           PSI
        </div>
        <div
          className="detail-value psi"
        >
          32
        </div>
        <div
          className="detail-state"
        >
          Good
        </div>
      </div>
      <div
        className="col-detail"
      >
        <div
          className="title-detail"
        >
           RAIN
        </div>
        <div
          className="detail-value"
        >
          0
        </div>
        <div
          className="detail-state"
        >
          mm
        </div>
      </div>
      <div
        className="col-detail"
      >
        <div
          className="title-detail"
        >
           DENGUE
        </div>
        <div
          className="circle-icon"
        />
      </div>
      <div
        className="col-detail"
      >
        <div
          className="add-icon"
        >
          <i
            className="far fa-plus-circle fa-2x"
          />
        </div>
        <div
          className="add-text"
        >
          Add
        </div>
      </div>
    </div>
  </WeatherDetail>
   
</AppProvider>
`;
